sqlmesh/core/engine_adapter/postgres.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import logging
+import typing as t
+
+from pandas.io.sql import read_sql_query
+from sqlglot import exp
+
+from sqlmesh.core.engine_adapter.base import EngineAdapterWithIndexSupport
+from sqlmesh.core.engine_adapter.base_postgres import BasePostgresEngineAdapter
+from sqlmesh.core.engine_adapter.shared import TransactionType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import DF, QueryOrDF
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresEngineAdapter(BasePostgresEngineAdapter, EngineAdapterWithIndexSupport):
+    DIALECT = "postgres"
+
+    @property
+    def cursor(self) -> t.Any:
```

Comment:
is this different from redshift?

Reply:
There is a difference. Redshift has this line: `cursor.paramstyle = "qmark"`. 

I did consider having them share a cursor in the base with the `autocommit = True` part but really they are two different libraries so I thought it would be best to leave this connection/cursor specific nuances in the two implementations.